actors/builtin/miner/miner_state.go

```diff
@@ -259,6 +272,129 @@ func (st *State) ComputeProvingSet(store adt.Store) ([]abi.SectorInfo, error) {
 	return sectorInfos, nil
 }
 
+func (st *State) addLockedFund(store adt.Store, currEpoch abi.ChainEpoch, pledgeAmount abi.TokenAmount) error {
+	Assert(pledgeAmount.GreaterThanEqual(big.Zero()))
+
+	lockedFundsArray := adt.AsArray(store, st.LockedFundsArray)
+
+	cliffStartEpoch := currEpoch + PledgeCliff
+	fullyVestedEpoch := currEpoch + PledgeCliff + PledgeVestingPeriod
+
+	for i := cliffStartEpoch; i < fullyVestedEpoch; i += VestIncrement {
+		var lockedFundEntry abi.TokenAmount
+		quantizedVestEpoch := getQuantizedEpoch(i)
+		found, err := lockedFundsArray.Get(epochKey(quantizedVestEpoch), &lockedFundEntry)
```

Comment:
lets add a TODO note here, these fancy ADT types do a full flush on every operation. We should avoid doing that.